This patch is based on the upstream commit below, adjusted for
the Debian package. It is mainly applied to clear the way for
later patches, not because it's likely to be useful in itself

commit f421eefd3444ef1d314e9af5b052e05fb3bdebd9
Author: Jubilee <46493976+workingjubilee@users.noreply.github.com>
Date:   Sat Sep 26 14:27:55 2020 -0700

    Rebuild CI stages (#295)
    
    Rebuilds CI into a set of stages that indicates first the "totally
    supported" stage, marked "build-test-verify", then groups arches
    by base pointer width and tier. This surfaces all targets which
    currently have full expected functionality, so as to better identify
    and limit future regressions.
    
    - Allow improper ctypes in project to quash spurious CI error
    - Remove Intel SDE, as the Dockerfile is broken.
    - Use recent LTS Ubuntu in CI
    - Fix deprecation warnings for .travis.yml
    - Allow inlining failure on 32-bit platforms
    - Always test aarch64 with Neon
    - Move promising platforms to build-test-verify
    - Drop tier 3 targets
    - Reduce repetitive SSE2 -> SSE4 tests
    - Allow aarch64-apple-ios + Neon test failure

Index: rust-packed-simd-0.3.3/src/lib.rs
===================================================================
--- rust-packed-simd-0.3.3.orig/src/lib.rs
+++ rust-packed-simd-0.3.3/src/lib.rs
@@ -212,16 +212,19 @@
     custom_inner_attributes
 )]
 #![allow(non_camel_case_types, non_snake_case,
-         clippy::cast_possible_truncation,
-         clippy::cast_lossless,
-         clippy::cast_possible_wrap,
-         clippy::cast_precision_loss,
-         // TODO: manually add the `#[must_use]` attribute where appropiate
-         clippy::must_use_candidate,
-         // This lint is currently broken for generic code
-         // See https://github.com/rust-lang/rust-clippy/issues/3410
-         clippy::use_self,
-         clippy::wrong_self_convention
+        // FIXME: these types are unsound in C FFI already
+        // See https://github.com/rust-lang/rust/issues/53346
+        improper_ctypes_definitions,
+        clippy::cast_possible_truncation,
+        clippy::cast_lossless,
+        clippy::cast_possible_wrap,
+        clippy::cast_precision_loss,
+        // TODO: manually add the `#[must_use]` attribute where appropriate
+        clippy::must_use_candidate,
+        // This lint is currently broken for generic code
+        // See https://github.com/rust-lang/rust-clippy/issues/3410
+        clippy::use_self,
+        clippy::wrong_self_convention,
 )]
 #![cfg_attr(test, feature(hashmap_internals))]
 #![deny(warnings, rust_2018_idioms, clippy::missing_inline_in_public_items)]
